csharp/BYOAuthFlow/Program.cs

```diff
@@ -0,0 +1,228 @@
+
+        /// <summary>
+        /// The Spatiald port to use for the local scenario
+        /// </summary>
+        private const int SpatialdPort = 9876;
+
+
+        /// <summary>
+        ///     PlEASE REPLACE ME.
+        ///     The SpatialOS Platform refresh token of a service account or a user account.
+        /// </summary>
+        private static string RefreshToken =>
+            Environment.GetEnvironmentVariable("IMPROBABLE_REFRESH_TOKEN") ?? "";
+
+        /// <summary>
+        ///     This contains the implementation of the "bring your own auth flow" scenario.
+        ///     1. Start a cloud deployment.
+        ///     2. Generate a Player Identity Token.
+        ///     3. List deployments.
+        ///     4. Generate a Login Token for a selected deployment.
+        ///     5. Connect to the deployment using the Player Identity Token and the Login Token.
+        /// </summary>
+        /// <param name="args"></param>
+        private static void Main(string[] args)
+        {
+            Console.WriteLine("Starting Cloud Scenario");
+            
+            var platformCredentials = new PlatformRefreshTokenCredential(RefreshToken);
+            RunScenario(
+                "locator.improbable.io",
+                443,
+                PlayerIdentityTokenServiceClient.Create(credentials: platformCredentials),
+                LoginTokenServiceClient.Create(credentials: platformCredentials),
+                DeploymentServiceClient.Create(credentials: platformCredentials),
+                insecureConnection: false);
+
+            Console.WriteLine("Starting Spatiald Scenario");
+            var spatialdEndpoint = new PlatformApiEndpoint("localhost", SpatialdPort, true);
+            RunScenario(
+                "localhost",
+                SpatialdPort,
+                PlayerIdentityTokenServiceClient.Create(spatialdEndpoint),
+                LoginTokenServiceClient.Create(spatialdEndpoint),
+                DeploymentServiceClient.Create(spatialdEndpoint),
+                insecureConnection: true);
+        }
+
+        private static void RunScenario(string locatorHost, ushort locatorPort, PlayerIdentityTokenServiceClient pitClient, LoginTokenServiceClient ltClient,
+            DeploymentServiceClient dsClient, bool insecureConnection)
+        {
+            var deployment = Setup(dsClient);
+            try
+            {
+                var pit = CreatePIT(pitClient);
+                deployment = FindDeployment(dsClient, deployment.Id);
+                var loginToken = CreateLoginTokenForDeployment(ltClient, pit, deployment.Id);
+                ConnectToLocator(locatorHost, locatorPort, insecureConnection, pit, loginToken);
+            }
+            finally
+            {
+                Cleanup(dsClient, deployment);
+            }
+        }
+
+        private static string CreatePIT(PlayerIdentityTokenServiceClient pitClient)
+        {
+            Console.WriteLine("Generate a PIT");
```

Comment:
Is there a reason why this (and a couple of others) has the verb in a different format ("Generate", "Create" instead of "Generating", "Creating")?

Reply:
Updated to Create